### PR TITLE
Quill Editor & Choice (Select2 style) Field

### PR DESCRIPTION
**Pull request by arifpavel**
1. Famous [quill](https://quilljs.com/) editor field. 

```
use Tanthammar\TallForms\QuillEditor;

QuillEditor::make('global.post.content', 'content')
                ->rules('required|string')
                ->allowMedia() //allow Media (image,video,url, fx) in toolbar - default is false
                ->theme('bubble') // default is 'snow'
                ->quillAdvanced() // add all the advanced quill options to the toolbar - default is false
                ->default('compose an awesome post...');
```

2. Choice (Select2 style field)
 
```
use Tanthammar\TallForms\Choice;
Choice::make('Choice')
        ->options($options)
        ->placeholder('Please select an option')
        ->rules(['required', Rule::in(array_values($options))]);
```

### CODE REVIEW

**Comment by tanthammar** [comment]
@arifpavel Cool.
I'll test this tomorrow.

Thank you for your contribution!

**Comment by arifpavel** [comment]
> @arifpavel Cool.
> I'll test this tomorrow.
> 
> Thank you for your contribution!

Okay, Hope these will add great value to this awesome package.

File: config/tall-forms.php [hunk 1]
```diff
@@ -28,6 +28,8 @@
     'multiselect-placeholder' => 'global.multiselect_placeholder', //'Please select one or multiple options ...'
     'multiselect-help' => 'global.multiselect_help', //'Press CTRL(Windows) or CMD(Mac), to select/deselect multiple options.'
 
+    //Choice placeholders and help
+    'choice-placeholder' => 'global.choice_placeholder', //'Please choice one or multiple options ...'
```
Comment (tanthammar) [inline config/tall-forms.php]: change the comment to

//'Please select one or multiple options ...'

**Comment by tanthammar** [comment]
I see now that this is actually two fields.
Can you please rename "Choice" field to "Choices"?
(Still haven't tested, just a quick change request)

**Review by tanthammar — COMMENTED** [review]
Please revert the changes made to Component and Field Select.php

**Comment by tanthammar** [comment]
Also have a general question about Choice.js, It has 145 unsolved issues.
I'm not very keen on supporting the field if it brings a lot of issues - are you up for the task?
I mean, can I assign issues to you?

File: config/tall-forms.php (hunk 1 above)
Comment (arifpavel) [reply config/tall-forms.php]: Okay

**Comment by arifpavel** [comment]
> Please revert the changes made to Component and Field Select.php

Sure, sorry for that (my ide did some indentation things :( )

**Comment by arifpavel** [comment]
> I see now that this is actually two fields.
> Can you please rename "Choice" field to "Choices"?
> (Still haven't tested, just a quick change request)

Choice currently supports single select only. (I've planned to make the Multiselect too). How about Choices (for multiselect)?

**Comment by arifpavel** [comment]
> Also have a general question about Choice.js, It has 145 unsolved issues.
> I'm not very keen on supporting the field if it brings a lot of issues - are you up for the task?
> I mean, can I assign issues to you?

Sure, I will try my best to support this. So, You can assign me related issues.

**Comment by tanthammar** [comment]
@arifpavel 
I messed up and closed this by mistake. 
Both fields look nice, but you have to make them work with the Reset button. 
Then I'll merge them :)

See `resources/views/components/trix.blade.php` for inspiration on how to respond to Livewire changes even though the field is `.defer`

Please reopen or make a new PR when you fixed that, and I thank you for contributing.

**Comment by arifpavel** [comment]
> @arifpavel
> I messed up and closed this by mistake.
> Both fields look nice, but you have to make them work with the Reset button.
> Then I'll merge them :)
> 
> See `resources/views/components/trix.blade.php` for inspiration on how to respond to Livewire changes even though the field is `.defer`
> 
> Please reopen or make a new PR when you fixed that, and I thank you for contributing.

No problem, I will work around the reset & will open a new PR. Thanks

**Comment by tanthammar** [comment]
@arifpavel How is it going?

**Comment by arifpavel** [comment]
> @arifpavel How is it going?

Sorry for the delay :( Actually loaded with lots of work & as Ramadan is going on couldn't manage time to work on this yet.

**Pull request merged**